tests/zero_code_change/tensorflow2_gradtape_integration_tests.py

```diff
@@ -0,0 +1,169 @@
+        n_epochs = 5
+        if script_mode:
+            if json_file_contents == "{}":
+                hook = smd.KerasHook(out_dir=sim.out_dir, export_tensorboard=True)
+            else:
+                hook = smd.KerasHook.create_from_json_file()
+
+            for epoch in range(n_epochs):
+                print("Epoch %d/%d" % (epoch + 1, n_epochs))
+                for data, labels in dataset:
+                    dataset_labels = labels
+                    labels = tf.one_hot(labels, depth=10)
+                    with hook.wrap_tape(tf.GradientTape()) as tape:
+                        logits = model(data, training=True)  # (32,10)
+                        loss_value = cce(labels, logits)
+                    grads = tape.gradient(loss_value, model.variables)
+                    opt.apply_gradients(zip(grads, model.variables))
+                    acc = train_acc_metric(dataset_labels, logits)
+                    hook.record_tensor_value(tensor_name="accuracy", tensor_value=acc)
+                log = "Epoch %d " % (epoch + 1)
+                log += "Accuracy %.4f" % train_acc_metric.result()
+                print(log)
+                train_acc_metric.reset_states()
+            hook = smd.get_hook()
+            assert hook
+            hook.close()
+            # Check that hook created and tensors saved
+            trial = smd.create_trial(path=sim.out_dir)
+            assert len(trial.steps()) > 0, "Nothing saved at any step."
+            assert len(trial.tensor_names()) > 0, "Tensors were not saved."
+            assert len(trial.tensor_names(collection="losses")) > 0
+        else:
+            # ZCC doesn't support yet (as of smdebug v0.7.2)
+            for epoch in range(n_epochs):
+                print("Epoch %d/%d" % (epoch + 1, n_epochs))
+                for data, labels in dataset:
+                    dataset_labels = labels
+                    labels = tf.one_hot(labels, depth=10)
+                    with tf.GradientTape(persistent=True) as tape:
+                        logits = model(data, training=True)  # (32,10)
+                        loss_value = cce(labels, logits)
+                    grads = tape.gradient(loss_value, model.variables)
+                    opt.apply_gradients(zip(grads, model.variables))
+                    acc = train_acc_metric(dataset_labels, logits)
+                log = "Epoch %d " % (epoch + 1)
+                log += "Accuracy %.4f" % train_acc_metric.result()
+                print(log)
+                train_acc_metric.reset_states()
+            hook = smd.get_hook()
+            assert not hook
+
+
+def test_keras_v2_default(script_mode: bool = False):
+    # Test default ZCC behavior
+    helper_test_keras_v2_gradienttape(script_mode=script_mode)
+
+
+def test_keras_v2_multi_collections(script_mode: bool = False):
+    # Test multiple collections included in hook json
+    json_file_contents = """
```

Comment:
`tests/zero_code_change/utils.py` has a build_json function.
Can it be used here?

Reply:
we'll have to modify the sagemaker simulator to make use of this build_json. maybe do it in a separate PR on master?